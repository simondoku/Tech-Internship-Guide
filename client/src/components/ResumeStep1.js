import React from "react";
import { Col, Row } from "react-bootstrap";
import Form from "react-bootstrap/Form";

export const ResumeStep1 = () => {
  return (
    <Form>
      <Form.Group className="mb-3" controlId="personal">
        <Row>
          <Col>
            <Form.Control placeholder="Full name" />
          </Col>

          <Col>
            <Form.Control type="email" placeholder="Email address" />
          </Col>
          <Col>
            <Form.Control type="mobile" placeholder="Mobile" />
          </Col>
        </Row>
      </Form.Group>

      <Form.Group className="mb-3" controlId="address">
        <Form.Control placeholder="Address" />
      </Form.Group>

      <Form.Group className="mb-3" controlId="skills">
        <Form.Control placeholder="Skills" />
      </Form.Group>
    </Form>
  );
};
