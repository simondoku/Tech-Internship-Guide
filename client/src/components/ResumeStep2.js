import React from "react";
import { Col, Row } from "react-bootstrap";
import Form from "react-bootstrap/Form";

export const ResumeStep2 = () => {
  return (
    <Form>
      <Form.Group className="mb-3" controlId="college1">
        <Row>
          <Col>
            <Form.Control placeholder="College name" />
          </Col>

          <Col>
            <Form.Control placeholder="Course" />
          </Col>
        </Row>
      </Form.Group>

      <Form.Group className="mb-3" controlId="marks1">
        <Row>
          <Col>
            <Form.Control placeholder="Percentage" />
          </Col>

          <Col>
            <Form.Control placeholder="Year of completion" />
          </Col>
        </Row>
      </Form.Group>
      <hr />
      <Form.Group className="mb-3" controlId="college2">
        <Row>
          <Col>
            <Form.Control placeholder="College name" />
          </Col>

          <Col>
            <Form.Control placeholder="Course" />
          </Col>
        </Row>
      </Form.Group>

      <Form.Group className="mb-3" controlId="marks2">
        <Row>
          <Col>
            <Form.Control placeholder="Percentage" />
          </Col>

          <Col>
            <Form.Control placeholder="Year of completion" />
          </Col>
        </Row>
      </Form.Group>
    </Form>
  );
};
