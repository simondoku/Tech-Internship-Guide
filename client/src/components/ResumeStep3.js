import React from "react";
import { Col, Row } from "react-bootstrap";
import Form from "react-bootstrap/Form";

export const ResumeStep3 = () => {
  return (
    <Form>
      <Form.Group className="mb-3" controlId="org1">
        <Row>
          <Col>
            <Form.Control placeholder="Organization" />
          </Col>

          <Col>
            <Form.Control placeholder="Position" />
          </Col>
        </Row>
      </Form.Group>

      <Form.Group className="mb-3" controlId="marks1">
        <Row>
          <Col>
            <Form.Control placeholder="From (eg. dd-mm-yyyy)" />
          </Col>

          <Col>
            <Form.Control placeholder="To (eg. dd-mm-yyyy)" />
          </Col>
        </Row>
      </Form.Group>
      <Form.Group className="mb-3" controlId="role1">
        <Row>
          <Col>
            <Form.Control placeholder="Role" />
          </Col>
        </Row>
      </Form.Group>
    </Form>
  );
};
