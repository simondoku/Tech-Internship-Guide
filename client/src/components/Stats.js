import React from "react";
import { Card } from "react-bootstrap";

export const Stats = () => {
  return (
    <div className="px-3 text-info border-top border-white">
      <div className="text-center mb-sm-5 mt-4">
        <h3 className="text-primary fs-2 fw-bold">Stats</h3>
      </div>
      <div
        style={{ display: "flex", justifyContent: "space-evenly", flex: ".8" }}
        className="m-4"
      >
        <Card>
          <Card.Body>
            <Card.Title>Total Visitors</Card.Title>
            <Card.Text className="text-center">500000</Card.Text>
          </Card.Body>
        </Card>

        <Card>
          <Card.Body>
            <Card.Title>Total Registrations</Card.Title>
            <Card.Text className="text-center">200000</Card.Text>
          </Card.Body>
        </Card>
      </div>
    </div>
  );
};
