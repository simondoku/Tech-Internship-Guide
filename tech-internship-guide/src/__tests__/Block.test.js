import React from "react";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import { Block, BlockSection } from "../components/Block";
import renderer from "react-test-renderer";
import { BLOCK_CONTENTS, BLOCK_SECTIONS } from "../core/block-cores";

// Clean up after each test
afterEach(cleanup);

describe("Block Component Testing", () => {
  // Test for rendering Block component with props
  test("Block component renders correctly with props", () => {
    // Arrange: Prepare props
    const blockId = Object.keys(BLOCK_CONTENTS)[0];
    const blockData = BLOCK_CONTENTS[blockId];

    // Act: Render the Block component
    render(<Block blockId={blockId} />);

    // Assert: Check if elements are present
    const blockNameElement = screen.getByText(blockData.name);
    const viewButton = screen.getByTestId("RemoveRedEyeOutlinedIcon");
    const launchButton = screen.getByTestId("LaunchIcon");

    expect(blockNameElement).toBeInTheDocument();
    expect(viewButton).toBeInTheDocument();
    expect(launchButton).toBeInTheDocument();
  });

  // Test for clicking on the view button to open a modal
  test("Clicking on view button opens a Modal", () => {
    const blockId = Object.keys(BLOCK_CONTENTS)[0];
    const blockData = BLOCK_CONTENTS[blockId];

    render(<Block blockId={blockId} />);

    const viewButton = screen.getByTestId("RemoveRedEyeOutlinedIcon");

    // Simulate a click on the view button
    fireEvent.click(viewButton);

    // Assert: Check if the modal content is present and matches with the block
    expect(screen.getByText(blockData.description)).toBeInTheDocument();
  });

  // Test for clicking on the launch button to open a new tab
  test("Clicking on launch button opens a new tab", () => {
    const blockId = Object.keys(BLOCK_CONTENTS)[0];
    const blockData = BLOCK_CONTENTS[blockId];

    render(<Block blockId={blockId} />);

    // Get all elements with role "link" (assuming this is how your launch button is implemented)
    const launchButton = screen.getAllByRole("link");

    // Get the last launch button (assuming it's the one you want to test)
    const lastLaunchButton = launchButton[launchButton.length - 1];

    // Assert: Check attributes of the link element
    expect(lastLaunchButton).toHaveAttribute("target", "_blank");
    expect(lastLaunchButton).toHaveAttribute("rel", "noreferrer");
    expect(lastLaunchButton).toHaveAttribute(
      "href",
      `https://www.${blockData.link}`
    );
  });

  test("Testing BlockSection Component exported from Block", () => {
    const sectionName = "Test Section";
    const sectionId = "pFundamentals";

    render(<BlockSection sectionName={sectionName} sectionId={sectionId} />);
    expect(screen.getByText(sectionName)).toBeInTheDocument();

    // pfundamentals is property of BLOCK_SECTION obj, we are using it to access another object BLOCK_CONTENTS
    BLOCK_SECTIONS[sectionId].forEach((element) => {
      // BLOCK_CONTENTS is taking element as 'key' to access the corresponding object mapped to it
      // and calling name property of the object
      // e.g. if element = w3school, then textElement = BLOCK_CONTENTS.w3school.name
      const textElement = BLOCK_CONTENTS[element].name;
      expect(screen.getByText(textElement)).toBeInTheDocument();
    });
  });

  // Snapshot testing of the Block component
  test("Create Snapshot of the Block", () => {
    // Arrage: Prepare Props
    const blockId = Object.keys(BLOCK_CONTENTS)[0];

    // Act: Create a snapshot of the Block component
    const component = renderer.create(<Block blockId={blockId} />).toJSON();

    // Assert: Compare the snapshot
    expect(component).toMatchSnapshot();
  });
});
