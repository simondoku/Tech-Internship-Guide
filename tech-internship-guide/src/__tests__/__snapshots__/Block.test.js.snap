// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Component Testing Create Snapshot of the Block 1`] = `
<div
  className="mb-2 text-light card"
  style={
    Object {
      "backgroundColor": "black",
      "color": "white",
      "width": "18rem",
    }
  }
>
  <div
    className="rounded"
    style={
      Object {
        "backgroundColor": "#eeeeee",
        "height": "9rem",
        "margin": "10px",
      }
    }
  >
    <img
      className="card-img-top img-fluid"
      src=""
    />
  </div>
  <div
    className="pb-1 px-2"
    style={
      Object {
        "display": "flex",
        "flex": "1",
        "justifyContent": "space-between",
      }
    }
  >
    <div>
      W3schools
    </div>
    <div>
      <button
        className="btn btn-dark btn-sm"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="RemoveRedEyeOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 6.5c3.79 0 7.17 2.13 8.82 5.5-1.65 3.37-5.02 5.5-8.82 5.5S4.83 15.37 3.18 12C4.83 8.63 8.21 6.5 12 6.5m0-2C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zm0 5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5-2.5-1.12-2.5-2.5 1.12-2.5 2.5-2.5m0-2c-2.48 0-4.5 2.02-4.5 4.5s2.02 4.5 4.5 4.5 4.5-2.02 4.5-4.5-2.02-4.5-4.5-4.5z"
          />
        </svg>
      </button>
      <a
        href="https://www.w3schools.com"
        rel="noreferrer"
        target="_blank"
      >
        <button
          className="btn btn-dark btn-sm"
          disabled={false}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="LaunchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </button>
      </a>
    </div>
  </div>
</div>
`;
